articles/20240819_dev_con_vs_traditional_dev.md

```diff
@@ -0,0 +1,67 @@
+
+### Pros of Traditional Development Environments
+
+#### 1. **Direct Access to Hardware**
+
+Traditional setups allow developers to interact directly with the hardware, without the abstraction layer that containers introduce. This direct access can lead to better performance, particularly in applications that require intensive processing or real-time capabilities.
+
+#### 2. **Simplicity and Familiarity**
+
+For many developers, traditional development environments are familiar and straightforward. There’s no need to learn new tools or concepts, making it easier for teams to maintain and troubleshoot the setup. This simplicity can be particularly advantageous for small projects or teams that don’t require the advanced capabilities of containers.
+
+### Cons of Traditional Development Environments
+
+#### 1. **Environment Drift**
+
+Over time, traditional development environments are prone to "environment drift," where different developers’ setups begin to diverge. This can lead to inconsistencies in how code behaves across different machines, making debugging and collaboration more challenging. Managing dependencies and ensuring consistency becomes a manual and error-prone task.
+
+#### 2. **Onboarding Challenges**
+
+Setting up a traditional development environment can be time-consuming, especially for complex projects. New developers often spend significant time configuring their environment before they can start contributing. This delay can be a bottleneck in fast-paced development cycles.
+
+#### 3. **Potential for Host System Pollution**
+
+Installing numerous dependencies directly on the host system can lead to clutter and conflicts, especially when working on multiple projects. Over time, this can make the system difficult to maintain and may require frequent cleanups or even a full system reinstall to resolve issues.
+
+### Conclusion
+
+Choosing between Dev Containers and traditional development environments depends on the specific needs of your project and team. Dev Containers offer significant advantages in consistency, onboarding, and system cleanliness but come with the trade-offs of added complexity and potential performance overhead. Traditional environments, while simpler and more familiar, are prone to issues like environment drift and host system pollution.
+
+For teams working on large, complex projects where consistency and collaboration are key, Dev Containers are likely the better choice. On the other hand, for smaller projects or teams that prioritize simplicity and direct hardware access, traditional development environments may still hold the edge.
```

Comment:
not only large teams, but all teams needing consistency and collaboration, also for example contributing to open source projects